### ISSUE TITLE
(Machine Learning) Classification approaches

### ISSUE THREAD

**Issue by Yash-10** [SEDSCelestiaBPGC/BCI#11]
We could start with the dataset currently in the `Datasets` folder (http://gigadb.org/dataset/view/id/100295) to get an overall idea.

There might be no particular order for experimentation, but we could start with:

-  Linear classifiers since they sometimes work great. For example, linear discriminant analysis (LDA) tries to maximize separation between the classes, here, *left* and *right*.
- KNNs, SVMs (linear or non-linear version), tree-based methods (I am slightly biased towards gradient boosted trees using, say, the [`XGBoost`](https://xgboost.readthedocs.io/en/latest/) library).
- I think there are some projection methods that could also help in classifying data points in a reduced dimensional space (eg: PCA, t-SNE, etc.) without the need to explicitly train and test using ML approaches. These could be semi-supervised or unsupervised and could be used to check if there is really any distinguishing factor between left and right classes.
    - This might be helpful to remove uninteresting features or reduce some noise since the datasets are generally high-dimensional.
- [`sklearn`](https://scikit-learn.org/stable/) might have some methods for unsupervised/semi-supervised methods, for example, k-means clustering.
